examples/medplum-photon-integration/src/bots/create-photon-prescription.test.ts

```diff
@@ -0,0 +1,215 @@
+    );
+  });
+});
+
+describe('getPhotonTreatmentId', () => {
+  const mockAuthToken = 'mock-auth-token';
+
+  test('Get treatment ID from MedicationKnowledge', async () => {
+    const medplum = new MockClient();
+    const mockMedicationCode: CodeableConcept = {
+      coding: [{ system: 'http://www.nlm.nih.gov/research/umls/rxnorm', code: '123456', display: 'Test Medication' }],
+    };
+
+    await medplum.createResource({
+      resourceType: 'MedicationKnowledge',
+      code: {
+        coding: [
+          { system: NEUTRON_HEALTH_TREATMENTS, code: 'photon-treatment-id' },
+          { system: 'http://www.nlm.nih.gov/research/umls/rxnorm', code: '123456' },
+        ],
+      },
+    });
+
+    const result = await getPhotonTreatmentId(mockAuthToken, medplum, mockMedicationCode);
+    expect(result).toBe('photon-treatment-id');
+  });
+
+  test.skip('Queries Photon if no MedicationKnowledge', async () => {
+    const medplum = new MockClient();
+    const mockMedicationCode: CodeableConcept = {
+      coding: [{ system: 'http://www.nlm.nih.gov/research/umls/rxnorm', code: '123456', display: 'Test Medication' }],
+    };
+
+    global.fetch = vi.fn().mockResolvedValue({
+      ok: true,
+      json: () => Promise.resolve({ data: { treatmentOptions: [{ medicationId: 'photon-treatment-id' }] } }),
+    });
+
+    const result = await getPhotonTreatmentId(mockAuthToken, medplum, mockMedicationCode);
+    expect(result).toBe('photon-treatment-id');
+  });
+
+  test.skip('No medication code provided', async () => {
+    const medplum = new MockClient();
+    await expect(getPhotonTreatmentId(mockAuthToken, medplum, undefined)).rejects.toThrow(
+      'Medication must have a code'
+    );
+  });
+
+  test.skip('No RxNorm or NDC code', async () => {
+    const medplum = new MockClient();
+    const mockMedicationCode: CodeableConcept = {
+      coding: [{ system: 'http://example.com/unknown-system', display: 'Unknown Medication' }],
+    };
+
+    await expect(getPhotonTreatmentId(mockAuthToken, medplum, mockMedicationCode)).rejects.toThrow(
+      'Could not find medication in Photon'
+    );
+  });
+});
+
+describe('getPhotonIdByCoding', () => {
+  test('Get Photon ID from MedicationKnowledge', async () => {
+    const medplum = new MockClient();
+    const mockMedicationCode: CodeableConcept = {
+      coding: [{ system: 'http://www.nlm.nih.gov/research/umls/rxnorm', code: '123456', display: 'Test Medication' }],
+    };
+
+    const medicationKnowledge = await medplum.createResource({
+      resourceType: 'MedicationKnowledge',
+      code: {
+        coding: [
+          { system: NEUTRON_HEALTH_TREATMENTS, code: 'photon-treatment-id' },
+          { system: 'http://www.nlm.nih.gov/research/umls/rxnorm', code: '123456', display: 'Test Medication' },
+        ],
+      },
+    });
+
+    const result = await getPhotonIdByCoding(medplum, mockMedicationCode);
+    expect(result).toBe('photon-treatment-id');
```

Comment:
Test is failing as searching for the MedicationKnowledge by code is returning undefined.